Based on 01070d43248fb97f3b2a08d780ae5a392573ce34 Mon Sep 17 00:00:00 2001
From: Chet Ramey <chet.ramey@case.edu>
Date: Fri, 18 Jul 2025 11:53:01 -0400
Subject: fix for `wait -n' in posix mode; fix for long messages in readline;
 fix for short reads by `source' builtin; fix for crash on RISC-V machines;
 fix for bad memory read when getopts is called twice with different sets of
 arguments

---
 builtins/evalfile.c           |   2 +

diff --git a/builtins/evalfile.c b/builtins/evalfile.c
index 1fee9880..3026c5f0 100644
--- a/builtins/evalfile.c
+++ b/builtins/evalfile.c
@@ -160,8 +160,10 @@ file_error_and_exit:
       nr = read (fd, string, file_size);
       if (nr >= 0)
 	string[nr] = '\0';
+#if 0
       if (nr != file_size)
 	nr = -1;		/* XXX - didn't get the whole file */
+#endif
     }
   else
     nr = zmapfd (fd, &string, 0);
